File: sub-system-core/src/main/java/org/github/core/logger/p6spy/P6SpyLogger.java
package org.github.core.logger.p6spy;

import com.p6spy.engine.spy.appender.MessageFormattingStrategy;
import static org.apache.commons.lang3.StringUtils.isNotBlank;

public class P6SpyLogger implements MessageFormattingStrategy {
  @Override
  public String formatMessage(int connectionId, String now, long elapsed, String category, String prepared, String sql, String url) {
    return isNotBlank(sql) ? "Consume Time：" + elapsed + " ms\n   JDBC URL：" + url + "\nExecute SQL：" + sql.replaceAll("\\s+", " ") + "\n" : "";
  }
}

File: sub-system-core/src/main/java/org/github/core/logger/p6spy/P6SpyLogger.kt
package org.github.core.logger.p6spy

import com.p6spy.engine.spy.appender.MessageFormattingStrategy

class P6SpyLogger : MessageFormattingStrategy {
    override fun formatMessage(connectionId: Int, now: String, elapsed: Long, category: String, prepared: String, sql: String?, url: String): String {
        return when {
            sql.isNullOrBlank().not() -> """Consume：$elapsed ms || URL：$url || SQL：${sql!!.replace("\\s+".toRegex(), " ")}"""

            else -> ""
        }
    }
}
